src/gui/editors/SongEditor.cpp

```diff
@@ -291,42 +297,40 @@ void SongEditor::loadSettings( const QDomElement& element )
 
 
 
+/*! \brief Return grid size as number of bars */
 float SongEditor::getSnapSize() const
 {
-	// 1 Bar is the third value in the snapping dropdown
-	int val = -m_snappingModel->value() + 3;
+	float snapSize = SNAP_SIZES[m_snappingModel->value()];
+
 	// If proportional snap is on, we snap to finer values when zoomed in
 	if (m_proportionalSnap)
 	{
-		val = val - m_zoomingModel->value() + 3;
+		// Finds the closest available snap size
```

Comment:
- Previously, it was possible to use finer or larger snap sizes by either zooming in with small snap size or zooming out with large snap size, but it's not possible now.
- I wonder why you have chosen nearest instead of rounding up/down.

Reply:
> * Previously, it was possible to use finer or larger snap sizes by either zooming in with small snap size or zooming out with large snap size, but it's not possible now.

Sorry, I think I don't understand. Do you mean that when ProportionalSnap is engaged the value should change while zooming in and out? Is the behaviour you expect any different as what you see in the gif? (I added values 1/32 and 1/64 to snapping values, because before it ended in 1/16, maybe that has something to do with your concern) 
![SnapingZoom](https://github.com/LMMS/lmms/assets/68785450/4f64aa95-be32-4987-8b53-a5337787d921)

>     * I wonder why you have chosen nearest instead of rounding up/down.
I have to say that this code was proposed by @allejok96, so I don't have the context as to why he chose that option. But comparing with master the behavior is not so different.

Reply:
> Sorry, I think I don't understand. Do you mean that when ProportionalSnap is engaged the value should change while zooming in and out?

I mean, on the current `master`, I can make the snap size to 64 bars by setting to 12.5%+8Bars with proportional snap enabled, or 1/64 bars by 1600%+1/4Bar or 400%+1/16Bar.

Reply:
> I wonder why you have chosen nearest instead of rounding up/down.

The values we are trying to round to are increasing exponentially, so we can't just `round()`. @superpaik had figured it out earlier using `pow()`, but I removed it due to segfault in my brain (and because it wouldn't support snap sizes of 1/3, which the new design has opened the possibility to). If there is a c++ function that rounds to the closest value in a list of values then that would be the best solution.

Reply:
> The values we are trying to round to are increasing exponentially, so we can't just `round()`

To clarify, I didn't mean `round()`, but minimum not less than/maximum not more than.

Reply:
I added a new variable PROPORTIONAL_SNAP_SIZES that hold bigger and smaller values than SNAP_SIZES, so the end result is similar to master (see gif, above this PR, below master). Is not exactly, but I can't see which one is better and why. To me, as long as they are consistent, both are ok.
![SnapingZoom](https://github.com/LMMS/lmms/assets/68785450/e7b7340c-233b-4275-ae8a-e16c5bf84c3a)